File: C#/TratamentoClima/PrincipalClima.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace SistemaClima 
{
    class Principal 
    {
        static List<Clima> dados = new List<Clima>();
        static string arquivo = Path.GetFullPath(Path.Combine(Directory.GetCurrentDirectory(), "data", "dadosClimaticos.csv"));
        
        static List<Clima> lista_verao = new List<Clima>();
        static List<Clima> lista_outono = new List<Clima>();
        static List<Clima> lista_inverno = new List<Clima>();
        
        static string path = arquivo;

        static void Main() 
        {
            CarregarDados();
            ExibirLista();
            GerarRelatorio();
        }

        static void CarregarDados() 
        {
            if (string.IsNullOrEmpty(path) || !File.Exists(path)) 
            {
                Console.WriteLine("Erro: CSV não encontrado.");
                return;
            }

            var linhas = File.ReadAllLines(path);
            for (int i = 1; i < linhas.Length; i++) 
            {
                var l = linhas[i];
                var col = l.Split(',');
                if (col.Length < 4) continue;

                int peso = 0;
                string p = col[3].ToLower().Trim();
                if (p == "muita") peso = 3;
                else if (p == "média") peso = 2;
                else if (p == "pouca") peso = 1;

                Clima clima = new Clima(col[0], col[1], col[2], peso);
                dados.Add(clima);
                
                if (clima.Mes == "Dezembro" || clima.Mes == "Janeiro" || clima.Mes == "Fevereiro" || clima.Mes == "Março")
                    lista_verao.Add(clima);
                else if (clima.Mes == "Abril" || clima.Mes == "Maio" || clima.Mes == "Junho" || clima.Mes == "Julho")
                    lista_outono.Add(clima);
                else
                    lista_inverno.Add(clima);
            }
        }

        static void ExibirLista() {
            Console.WriteLine("\n--- DADOS CLIMÁTICOS ---");
            foreach (var item in dados) 
            {
                Console.WriteLine(item);
            }
            Console.WriteLine($"Total de registros: {dados.Count}\n");
        }

        static void GerarRelatorio() 
        {
            int chuvaV = 0, chuvaO = 0, chuvaI = 0;
            int quenteV = 0, quenteO = 0, quenteI = 0;
            int amenoV = 0, amenoO = 0, amenoI = 0;

            foreach (var item in lista_verao) 
            {
                chuvaV += item.Precipitacao;
                if (item.Temperatura.Equals("Quente", StringComparison.OrdinalIgnoreCase)) quenteV++;
                if (item.Temperatura.Equals("Ameno", StringComparison.OrdinalIgnoreCase)) amenoV++;
            }
            
            foreach (var item in lista_outono) 
            {
                chuvaO += item.Precipitacao;
                if (item.Temperatura.Equals("Quente", StringComparison.OrdinalIgnoreCase)) quenteO++;
                if (item.Temperatura.Equals("Ameno", StringComparison.OrdinalIgnoreCase)) amenoO++;
            }
            
            foreach (var item in lista_inverno) 
            {
                chuvaI += item.Precipitacao;
                if (item.Temperatura.Equals("Quente", StringComparison.OrdinalIgnoreCase)) quenteI++;
                if (item.Temperatura.Equals("Ameno", StringComparison.OrdinalIgnoreCase)) amenoI++;
            }

            string maisChuva = "Inverno";
            if (chuvaV >= chuvaO && chuvaV >= chuvaI) maisChuva = "Verão";
            else if (chuvaO >= chuvaV && chuvaO >= chuvaI) maisChuva = "Outono";

            string menosChuva = "Inverno";
            if (chuvaV <= chuvaO && chuvaV <= chuvaI) menosChuva = "Verão";
            else if (chuvaO <= chuvaV && chuvaO <= chuvaI) menosChuva = "Outono";

            string maisQuente = "Inverno";
            if (quenteV >= quenteO && quenteV >= quenteI) maisQuente = "Verão";
            else if (quenteO >= quenteV && quenteO >= quenteI) maisQuente = "Outono";

            string maisAmeno = "Inverno";
            if (amenoV >= amenoO && amenoV >= amenoI) maisAmeno = "Verão";
            else if (amenoO >= amenoV && amenoO >= amenoI) maisAmeno = "Outono";
        
            Console.WriteLine("--- RESULTADOS DO CLIMA ---");
            Console.WriteLine($"Estação mais chuvosa:  {maisChuva.ToUpper()}");
            Console.WriteLine($"Estação menos chuvosa: {menosChuva.ToUpper()}");
            Console.WriteLine($"Estação mais quente:   {maisQuente.ToUpper()}"); 
            Console.WriteLine($"Estação mais amena:    {maisAmeno.ToUpper()}");
            Console.WriteLine("---------------------------\n");
        }

    }
}

File: 01-listas/Python/Atividades-py/principal_clima.py
import os # importa o módulo os para manipulação de arquivos e diretórios
from clima import Clima 

class Principal:
    def __init__(self):
        self.dados = [] 
        pasta_script = os.path.dirname(os.path.abspath(__file__)) 
        pasta_projeto = os.path.abspath(os.path.join(pasta_script, "..", ".."))
        self.arquivo = os.path.join(pasta_projeto, "data", "dadosClimaticos.csv")
        
        # listas para armazenar os objetos da classe Clima separados por estação do ano
        self.lista_verao = []
        self.lista_outono = []
        self.lista_inverno = []
        
    def carregar_dados(self):
        """_summary_
        metodo que carrega os dados do arquivo CSV, criando objetos da classe Clima e distribuindo-os em listas por estação do ano
        """
        path = self.arquivo
        
        if not path or not os.path.exists(path):
            print("Erro: CSV não encontrado.")
            return
        
        with open(path, "r", encoding="utf-8") as arq:
            linhas = arq.read().splitlines()
        
        for linha in linhas[1:]:
            li = linha.split(",")
            if len(li) < 4:
                continue
            
            ano = li[0].strip() 
            mes = li[1].strip()
            temp = li[2].strip()
            chuva_txt = li[3].strip().lower() 
            
            peso = 0
            if chuva_txt == "muita":
                peso = 3
            elif chuva_txt == "média":
                peso = 2
            elif chuva_txt == "pouca":
                peso = 1
            
            clima = Clima(ano, mes, temp, peso)
            self.dados.append(clima)
            
            if mes in ["Dezembro", "Janeiro", "Fevereiro", "Março"]:
                self.lista_verao.append(clima)
            elif mes in ["Abril", "Maio", "Junho", "Julho"]:
                self.lista_outono.append(clima)
            else:
                self.lista_inverno.append(clima)
    
    def exibir_lista(self):
        print("\n--- DADOS CLIMÁTICOS ---")
        for item in self.dados:# itera sobre a lista de dados climáticos e exibe cada registro formatado usando o método __str__ da classe Clima
            print(item)
        print(f"Total de registros: {len(self.dados)}\n")

    
    def analisar(self):
        """_summary_
        método que analisa os dados climáticos, calculando a estação mais chuvosa, a estação menos chuvosa, a estação mais quente e a estação mais amena, e exibe os resultados formatados
        """
        chuva_verao = sum(item.precipitacao for item in self.lista_verao) 
        chuva_outono = sum(item.precipitacao for item in self.lista_outono)
        chuva_inverno = sum(item.precipitacao for item in self.lista_inverno)
        
        quente_verao = sum(1 for item in self.lista_verao if item.temperatura.lower() == "quente")
        quente_outono = sum(1 for item in self.lista_outono if item.temperatura.lower() == "quente")
        quente_inverno = sum(1 for item in self.lista_inverno if item.temperatura.lower() == "quente")
        
        ameno_verao = sum(1 for item in self.lista_verao if item.temperatura.lower() == "ameno")
        ameno_outono = sum(1 for item in self.lista_outono if item.temperatura.lower() == "ameno")
        ameno_inverno = sum(1 for item in self.lista_inverno if item.temperatura.lower() == "ameno")
        
        chuva = {"Verão": chuva_verao, "Outono": chuva_outono, "Inverno": chuva_inverno}
        quente = {"Verão": quente_verao, "Outono": quente_outono, "Inverno": quente_inverno}
        ameno = {"Verão": ameno_verao, "Outono": ameno_outono, "Inverno": ameno_inverno}
        
        mais_chuva = max(chuva, key=chuva.get)
        menos_chuva = min(chuva, key=chuva.get)
        mais_quente = max(quente, key=quente.get)
        mais_ameno = max(ameno, key=ameno.get)
        
        print("\n--- RESULTADOS DO CLIMA ---")
        print(f"Estação mais chuvosa:  {mais_chuva.upper()}")
        print(f"Estação menos chuvosa: {menos_chuva.upper()}")
        print(f"Estação mais quente:   {mais_quente.upper()}")
        print(f"Estação mais amena:    {mais_ameno.upper()}")
        print("---------------------------\n")
    
    def executar(self):
        self.carregar_dados()
        if self.dados:
            self.exibir_lista()
            self.analisar()


if __name__ == "__main__":
    sistema = Principal()
    sistema.executar()